stanford-sw/src/edu/stanford/StanfordIndexer.java

```diff
@@ -2380,12 +2438,47 @@ private void addSDRfrom856s(final Record record) {
     Set<String> resultSet = new LinkedHashSet<String>();
     for (String val : MarcUtils.getFieldList(record, fieldSpec)) {
         String result = Utils.removeAllTrailingCharAndPeriod(val, "(" + charsToReplaceRegEx + ")+", charsB4periodRegEx);
+        result = cleanFacetPunct(result);
       resultSet.add(result);
     }
 
     return resultSet;
   }
 
+    public String cleanFacetPunct(String origStr)
```

Comment:
I wasn't sure if it was okay to put things in core/src/org/solrmarc/tools/Utils.java. Is that what you meant by "Utils"?